src/lib/helper/buildPaginationHelper.ts

```diff
@@ -0,0 +1,162 @@
+import { Association } from "../../types/associations";
+import { LANDING_URL, ITEMS_PER_PAGE } from "../../constants";
+import { PageItem, PaginationData } from "../../types/pagination";
+
+export const buildPaginationElement = (
+    currentPageNumber: number,
+    numOfPages: number,
+    urlPrefix: string,
+    searchQuery: string
+): PaginationData => {
+    const pagination: PaginationData = { items: [] };
+    const pageItems: PageItem[] = [];
+
+    if (numOfPages <= 1 || currentPageNumber < 1) return pagination;
+
+    // Add Previous and Next
+    if (currentPageNumber !== 1) {
+        pagination.previous = {
+            href: `${urlPrefix}?page=${currentPageNumber - 1}${searchQuery}`
+        };
+    }
+    if (currentPageNumber !== numOfPages) { pagination.next = { href: `${urlPrefix}?page=${currentPageNumber + 1}${searchQuery}` }; }
+
+    // Add first element by default
+    pageItems.push(createPageItem(1, currentPageNumber, false, urlPrefix, searchQuery));
+
+    // Add second element if applicable - possible ellipsis
+    if (numOfPages >= 3) {
+        const isEllipsis = numOfPages >= 5 && currentPageNumber >= 5;
+        pageItems.push(createPageItem(2, currentPageNumber, isEllipsis, urlPrefix, searchQuery));
+    }
+
+    // Add element at middle left position if applicable
+    if (
+        numOfPages >= 5 &&
+    currentPageNumber >= 4 &&
```

Comment:
Please fix indentations here and in the following lines through the file.

Reply:
indentation has been fixed